org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/git/GitProvider.xtend

```diff
@@ -24,11 +38,20 @@ class GitProvider {
 	/**
 	 * @return the potentially cached {@link Git} instance for the current workspace.
 	 */
+	 
 	def Git getGit() {
```

Comment:
does it make sense to maybe make this method private or protected? Right now, the "wrong" (unconfigured) pull and push commands may easily be used, by accident.

I assume that setting the SSH session factory during initialization (i.e. in this classes' `initialize`/`initializeNew`/`reinitializeExisting` methods) is insufficient? It happens in `initializeNew`, I see, but seems to be local to the `init` command. 

Looks like the JGit API is a bit annoying in this regard… :\
A **Decorator** on class `Git` could work, though.